src/app/features/course-info/course-info.component.ts

```diff
@@ -1,13 +1,34 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, OnInit } from '@angular/core';
 import { Course } from '@shared/models/course.model';
 import { ButtonText } from '@shared/components';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CoursesStoreService } from '@app/services/courses-store.service';
 
 @Component({
   selector: 'app-course-info',
   templateUrl: './course-info.component.html',
   styleUrls: ['./course-info.component.scss'],
 })
-export class CourseInfoComponent {
-  @Input() course!: Course;
+export class CourseInfoComponent implements OnInit {
+  @Input() course!: Course | null;
   protected readonly ButtonText = ButtonText;
+
+  constructor(
+    private route: ActivatedRoute,
+    private routes: Router,
+    private coursesStoreService: CoursesStoreService
+  ) {}
+
+  ngOnInit() {
+    const courseId = this.route.snapshot.paramMap.get('id');
+    if (!courseId) return;
```

Comment:
Maybe navigate back to course list once we have a 404 on the course, shall we?
![image](https://github.com/user-attachments/assets/54f50779-514f-4e09-8a54-98a16ad9e65c)

Reply:
Yeah you're right